hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -2754,10 +2754,10 @@ protected PrepareFlushResult internalPrepareFlushCache(WAL wal, long myseqid,
           if (wal != null) {
             writeEntry = mvcc.begin();
             long flushOpSeqId = writeEntry.getWriteNumber();
+            mvcc.completeAndWait(writeEntry);
             FlushResultImpl flushResult =
                 new FlushResultImpl(FlushResult.Result.CANNOT_FLUSH_MEMSTORE_EMPTY, flushOpSeqId,
                     "Nothing to flush", writeFlushRequestMarkerToWAL(wal, writeFlushWalMarker));
-            mvcc.completeAndWait(writeEntry);
```

Comment:
if `writeEntry` be threadLocal,  the logic here is wrong. 
`writeFlushRequestMarkerToWAL` contains mvcc begin and complete